### DISCUSSION TITLE
Simple cantilever bar displacement not zero on fixed constrained node. (Isoparametric coord)

### DISCUSSION THREAD

**Discussion by devel0** [BriefFiniteElementNet/BriefFiniteElement.Net#156] [General]
Follow image generated through [this library example](https://github.com/devel0/netcore-opengl/blob/91e14e52e6de3740006d450020bf09d5792ad5bc/examples/example-0023/Program.cs#L17) and the model is base on [simple cantilever](https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/blob/b7173d2e3ab8e3896f365d5f1c5ee0943a3d307d/Samples/Examples.CSharp/SimpleCantilever.cs) with the only difference within the nodal force horizontalLoad increased of 1e2 factor.

Shouldn't the beam global displacement at offset=0 where there is a fixed constrained node a zero value ?
thanks you for any hint

![image](https://user-images.githubusercontent.com/13405008/230639230-a40d4178-55df-4691-b5e6-d28bc7cf09bd.png)

**Comment by epsi1on** [comment]
Hi,
recently the nuget packages are updated, is there same problem with older nuget packages like [2.0.0](https://feed.nuget.org/packages/BriefFiniteElement.Net/2.0.0-pre)?

**Comment by epsi1on** [comment]
also if you could make a source code that i can run on my side and check, it also would be helpful

**Comment by devel0** [comment]
of course, this is a console mode repro ( btw note that netcore-opengl can run in linux and even windows net7 ),
following the commands to clone repro repository at the branch used for this repro and to run using dotnet ( https://dotnet.microsoft.com/en-us/download ):

```sh
git clone --branch brief-finite-element/cantilever-fixed-restraint git@github.com:devel0/repros.git
cd repros/cantilever-fixed-restraint
dotnet run
```

from visual studio open .csproj or from vscode open folder and accept to create assets to run the solution.
If haven't net7 could downgrade from .csproj target framework.

I tried using 2.1.0 nuget package (latest) getting same results as described in the issue.

**Comment by epsi1on** [comment]
this line of code should be changed:
``
System.Console.WriteLine($"beam disp at off=0 dx = {((BarElement)model.Elements[0]).GetGlobalDisplacementAt(0).DX}");
``
to
``
System.Console.WriteLine($"beam disp at off=0 dx = {((BarElement)model.Elements[0]).GetGlobalDisplacementAt(-1.0).DX}");
``

the coordination you pass to ``Element.GetGlobalDisplacementAt(double xi)`` is isoparametric coordination, and it is not local coordination of element.
It means start of element is `x=0` and `ξ=-1`, middle of element is `x=L/2` and `ξ=0` and end of element is `x=L` and `ξ=+1`.
Note that `xi` do not mean `x[i]`, it is greek letter `ξ` (pronounce kesi or zi or xi).
For more info see [this section](https://bfenet.readthedocs.io/en/latest/elements/finiteElements/Bar/coords.html#iso-parametric-coordination-system-for-barelement-with-two-nodes)
you can transform between `ξ` and `x` as described in above link, also in `BarElement.IsoCoordsToLocalCoords()` and `BarElement.LocalCoordsToIsoCoords()`

**Comment by devel0** [comment]
Many thanks,

I adjusted [the example](https://github.com/devel0/netcore-opengl/blob/6fe8d8af4b9c43cf2ea881e9919b8a1991a32d3b/examples/example-0023/Program.cs#L144) and now it display data as expected.

![image](https://user-images.githubusercontent.com/13405008/230718135-3179f71f-266c-4d16-a4e5-6a52d4503af7.png)
